src/listpack.c

```diff
@@ -369,6 +369,23 @@ static inline unsigned long lpEncodeBacklen(unsigned char *buf, uint64_t l) {
     }
 }
 
+/* Calculate the number of bytes required to reverse-encode a variable length
+ * field representing the length of the previous element of size 'l', ranging
+ * from 1 to 5. */
+static inline unsigned long lpEncodeBacklenBytes(uint64_t l) {
+    if (l <= 127) {
+        return 1;
+    } else if (l < 16383) {
```

Comment:
This is unrelated to your changes, but why are these `<` and not `<=`? It looks like as long as the left-most 1 bit stays in place(it will move left by 1 index for 16384) the limit number(16383) can still be encoded in 2 bytes. And the same is valid for the other else-ifs below.

Reply:
@ilitov you're right, in this case (16384, 2097152) we would be wasting 1 byte and fixing it would not be cost-effective to fix it because we need to upgrade the rdb version to compact the old listpack structure.